### ISSUE TITLE
Support for Android private / signature permissions

### ISSUE THREAD

**Issue by Qwin** [zoontek/react-native-permissions#887]
### Why it is needed?

Currently the Permission type does not support private/signature permissions like com.google.android.gms.permission.CAR_VENDOR_EXTENSION, I would like to ask to support those permissions as well or at least just allow freeform as an overload. 

### Possible implementation

requestMultiple([PERMISSIONS.ANDROID.BLUETOOTH_CONNECT, PERMISSIONS.ANDROID.BLUETOOTH_SCAN, "com.google.android.gms.permission.CAR_VENDOR_EXTENSION"])

just a note but the above does not work sadly I get unavailable and never get a prompt when I forcibly type it as Permission, while using AndroidPermissions (default react-native library) does work with the above code for the private permission

**UPDATE**:

looking at the code a bit closer I think the issue is with this method here not accounting for privledged / signature permissions:
<img width="891" alt="image" src="https://github.com/user-attachments/assets/9a3f128d-be9f-46ae-99c6-b6c9b82e85c7">

**UPDATE 2**

So figured out the issue it seems that https://developer.android.com/reference/android/Manifest.permission doesnt have the CAR_VENDOR_EXTENSION permission in it due to it being a system level permission only given out to some companies. Because of that I propose the following solution:

```
requestMultiple(permission, { skipManifestCheck: true }), that way you can skip the check above and avoid it from being unavailable. 
```

**UPDATE 3**

looking at the permission itself based on this post, it seems to be a dangerous permission:
https://stackoverflow.com/questions/32061034/how-do-we-list-all-defined-permissions

**Comment by zoontek** [comment]
@Qwin This is not an issue anymore [in v5.0.0](https://github.com/zoontek/react-native-permissions/pull/890#issuecomment-2346411328), as the `UNAVAILABLE` status has been removed (merged with `BLOCKED`) - like the builtin `PermissionsAndroid` module does with `NEVER_ASK_AGAIN`).

You need to cast the type:

```ts
check("com.google.android.gms.permission.CAR_VENDOR_EXTENSION" as Permission)
```

**Comment by zoontek** [comment]
This is fixed in [v5.0.0](https://github.com/zoontek/react-native-permissions/releases/tag/5.0.0). As it's not exposed, you need to cast it:

```ts
requestMultiple("com.google.android.gms.permission.CAR_VENDOR_EXTENSION" as Permission)
```
